src/capi/azext_capi/custom.py

```diff
@@ -309,9 +369,34 @@ def create_workload_cluster(  # pylint: disable=unused-argument,too-many-argumen
         ephemeral_disks=False,
         windows=False,
         pivot=False,
-        output_path=None,
+        user_provided_template=None,
         yes=False):
 
+    if user_provided_template:
+        mutual_exclusive_args = [
+            {
+                "name": "external_cloud_provider",
+                "value": external_cloud_provider
+            },
+            {
+                "name": "machinepool",
+                "value": machinepool
+            },
+            {
+                "name": "ephemeral_disks",
+                "value": ephemeral_disks
+            },
+            {
+                "name": "windows",
+                "value": windows
```

Comment:
I was testing this out and found that this is a flag inside the built in templates but it is also used to deploy the CNI for windows: https://github.com/Azure/azure-capi-cli-extension/blob/86d2687f081ca3bfcee7dbc21110daf31ef34c69/src/capi/azext_capi/custom.py#L437

Reply:
we will need this if we want to deploy windows cni (which I think we do), the user can opt into deploying it

Reply:
Definitely, I was looking at it and having the option for the user to opt in will be good. @mboersma and @jsturtevant, should we add a flag like `--windows-cni`  to prevent overloading `--windows` flag purpose? or We could also only keep single flag and update the help command, what do you think?

Reply:
imo we can just re-use `--windows` flag. be curious what @mboersma thinks

Reply:
I think we can re-use the `--windows` flag. `az capi` can be opinionated about things. In fact once #99 merges, users have a way to accomplish basically any customization, so I think that's an argument for keeping `az capi create` simple otherwise.